sklearn/linear_model/coordinate_descent.py

```diff
@@ -712,7 +712,7 @@ def fit(self, X, y, check_input=True):
 
         X, y, X_offset, y_offset, X_scale, precompute, Xy = \
             _pre_fit(X, y, None, self.precompute, self.normalize,
-                     self.fit_intercept, copy=False)
+                     self.fit_intercept, copy=self.copy_X)
```

Comment:
this is not the right fix as if you want copy_X=True with check_input=True you will make 2 copies: one with check_input and one here. That's why it is set to False here.